### PR TITLE
Standard addVariable and addFactor API POC

### PR DESCRIPTION
**Pull request by Affie**
This is a POC for discussion for `addVariable` and `addFactor` wrappers to follow a service bridge model with signatures based on the current IIF signatures.  Basic usage will be:

This is no longer part of this pr
```julia
nva = NVA.NavAbilityPlatform()
addVariable(nva, "x0", :Position1)
addFactor(nva, ["x0"], NVA.Prior(Z=Normal(0.,1.)))
```

Client context will remain seperate as:
```julia
addVariable(client, context,  "x0", :Position1)
addFactor(client, context, ["x0"], NVA.Prior(Z=Normal(0.,1.)))
```

The changes that are up for discussion are mostly grouped in "src/navability/services/StandardAPI.jl"
I hope this is enough detail to show the intent. If it's on the right path I'll do the cleanup and tests in full.
I will also implement it in python if this is ok. 

Notes
- Standard API for `addVariable` and client_context from #78 
- The FactorData approach follows the py sdk more, see also #74
- The naming of the client content issue is here https://github.com/NavAbility/nva-sdk/issues/23
    - I just went with the name in #78 for now

The [**Data kwargs](https://github.com/NavAbility/NavAbilitySDK.jl/commit/1cc607162329d0355f31b27be07df84c18aaa654) commit is for supporting nullhypo #108 and multihypo and we can just cherry-pick it in a separate PR if needed.

### CODE REVIEW

**Comment by dehann** [comment]
HI @Affie , awesome thanks -- I will look at this later in the week, just want to get past work for Thursday.  I'd like to get these improvements in well before ICRA. Both on SDK.jl and SDK.py.

**Comment by Affie** [comment]
Updated for discussion captured in https://github.com/NavAbility/nva-sdk/issues/21#issuecomment-1105501578

**Comment by Affie** [comment]
Ok, I've added a few more types and (very) basic tests. PR is ready for imputs.

**Comment by dehann** [comment]
Short of replicating in other SDKs, this PR closes for me:
- https://github.com/NavAbility/nva-sdk/issues/21

File: src/navability/entities/InferenceTypes.jl [hunk 1]
```diff
@@ -8,7 +8,23 @@ functions inside of factors.
 """
 abstract type InferenceType end
 
-#TODO maybe generate all ZInferenceTypes with a macro
+macro nvaZInferenceType(inferencetype)
+  return esc(quote
+      Base.@__doc__ struct $inferencetype <: NVA.InferenceType
```
Comment (dehann) [inline src/navability/entities/InferenceTypes.jl:13]: is that the same as `Base.@kwdef`?

Comment (Affie) [reply src/navability/entities/InferenceTypes.jl:13]: No, that part is covered by line 16.
This macro is just to reduce code maintenance a bit with standard Z field types.

**Comment by Affie** [comment]
Another difference between python and julia SDK's are in the naming of inference types.
For example, Python uses just: Pose2Point2BearingRange
while julia has `Pose2Point2BearingRangeInferenceType`, but then uses additional functions to construct the data i.e. `Pose2Point2BearingRangeData`

**Comment by dehann** [comment]
FYI, im going to merge the new (not exported) `addData` `getData` ... PRs before merging this PR.  They will likely all land in next patch release.

**Review by dehann — APPROVED** [review]
looks good thanks!  I'll merge in a few hours.

**Comment by dehann** [comment]
was just able to run through Tut 1-4 on this branch, all looks good.  There are a few things in the notebooks that I want to improve, but this PR and tagging is ready to go.

**Pull request merged**